californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/LockstepEndpoint.java

```diff
@@ -572,17 +669,99 @@ public EmptyMessageExpectation(Type type, int mid) {
 		}
 
 		@Override
+		public void go(Message msg) throws Exception {
+			if (CoAP.isEmptyMessage(msg.getRawCode())) {
+				check(msg);
+			} else {
+				Assert.fail("Expected empty message for " + this + ", but received " + msg);
+			}
+		}
+
+		@Override
+		public void add(MultiMessageExpectation multi) {
+			multi.add(this);
+		}
+	}
+
+	public class MultiMessageExpectation implements Action {
+
+		private int counter;
+		private EmptyMessageExpectation emptyExpectation;
```

Comment:
is there a particular reason why we only allow one message per type? Couldn't these be Lists instead?